src/dms/core/EdFi.DataManagementService.Core/Middleware/JwtRoleAuthenticationMiddleware.cs

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Security.Claims;
+using System.Text.Json;
+using EdFi.DataManagementService.Core.External.Model;
+using EdFi.DataManagementService.Core.Model;
+using EdFi.DataManagementService.Core.Pipeline;
+using EdFi.DataManagementService.Core.Security;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace EdFi.DataManagementService.Core.Middleware;
+
+/// <summary>
+/// Middleware for JWT role-based authentication in the DMS Core pipeline.
+/// This middleware validates JWT tokens and checks for required roles without extracting ClientAuthorizations.
+///
+/// The purpose of this middleware is to provide authentication for non-CRUD resource endpoints.
+/// </summary>
+internal class JwtRoleAuthenticationMiddleware(
```

Comment:
A companion to the other JWT Auth middleware, this one is a bit speculative and is for endpoints that we want to secure but don't use ClientAuthorizations (like a management endpoint or something)